### PR TITLE
feat: add user create & update webhook

### PR DESCRIPTION
**Pull request by johnytiago**

### CODE REVIEW

File: apps/asap-server/src/cms/users.ts [hunk 1]
```diff
@@ -114,4 +114,20 @@ export default class Users extends Base {
...
+  updateOrcidWorks(
+    user: CMSUser,
+    lastModifiedDate: string,
+    works: CMSOrcidWork[],
+  ): Promise<CMSUser> {
+    return this.client
+      .patch<CMSUser>(`users/${user.id}`, {
+        json: {
+          email: { iv: user.data.email.iv },
+          orcidLastModifiedDate: { iv: lastModifiedDate },
```
Comment (fampinheiro) [inline apps/asap-server/src/cms/users.ts:127]: Is the last modified date from the orcid system or defined by us?

File: apps/asap-server/src/handlers/users/sync-orcid.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      if (newOrcid && newOrcid !== payload.dataOld?.orcid?.iv) {
+        return {
+          statusCode: 200,
+          payload: await users.syncOrcidProfile(id),
+        };
+      }
+      return { statusCode: 204 };
+    }
+
+    throw Boom.notImplemented();
```
Comment (fampinheiro) [inline apps/asap-server/src/handlers/users/sync-orcid.ts:58]: Can you check if we need to return 200? I'm assuming a non-success code will make a new request to the lambda.

Comment (johnytiago) [reply apps/asap-server/src/handlers/users/sync-orcid.ts:58]: To be safe, I'm going to return a 204 instead 👍

File: apps/asap-server/src/cms/users.ts (hunk 1 above)
Comment (johnytiago) [reply apps/asap-server/src/cms/users.ts:127]: This one come from the orcid system

File: apps/asap-server/src/entities/user.ts [hunk 3]
```diff
@@ -6,17 +6,32 @@ export interface CMSUser {
   data: {
     displayName: { iv: string };
     email: { iv: string };
-    firstName: { iv: string };
-    middleName: { iv: string };
-    lastName: { iv: string };
-    jobTitle: { iv: string };
-    orcid: { iv: string };
-    institution: { iv: string };
+    firstName?: { iv: string };
```
Comment (fampinheiro) [inline apps/asap-server/src/entities/user.ts:9]: Is there any particular reason to make this optional? We should consider redefining the type for the particular use-case with need this flexibility.

File: apps/asap-server/test/helpers/create-user.ts [hunk 4]
```diff
@@ -26,12 +26,12 @@ function transform(user: CMSUser): TestUserResponse {
...
     orcid: user.data.orcid && user.data.orcid.iv,
     institution: user.data.institution && user.data.institution.iv,
-    teams: user.data.teams,
+    teams: user.data.teams?.iv,
     connections: user.data.connections.iv,
   };
 }
 
-export const createUser = (): Promise<CMSUser> => {
+export const createUser = (overwrites: Invitee | {}): Promise<CMSUser> => {
```
Comment (fampinheiro) [inline apps/asap-server/test/helpers/create-user.ts:34]:
```suggestion
export const createUser = (overwrites: Invitee | object = {}): Promise<CMSUser> => {
```

File: apps/asap-server/src/entities/user.ts (hunk 3 above)
Comment (johnytiago) [reply apps/asap-server/src/entities/user.ts:9]: I just updated it to reflect what is mandatory in the createSchema.

File: apps/asap-server/test/helpers/create-user.ts (hunk 4 above)
Comment (johnytiago) [reply apps/asap-server/test/helpers/create-user.ts:34]: 🤦

**Pull request merged**